### PR TITLE
Extract runner from unmock-core

### PR DESCRIPTION
**Pull request by carolstran**
Part of #299 

- Created separate `unmock-runner` package
- Made exports more generic to be able to run with Jest, Mocha or any other future imported runner function
- Added a temporary `jestRunner` function under `unmock-runner` to use until we have a proper `unmock-jest-runner` package

TODO in future PRs: 
- Publish this as a package on NPM 
- Create/publish `unmock-jest-runner` (to replace `packages/unmock-runner/src/jestRunner`) 
- Create/publish `unmock-mocha-runner`
- Add appropriate documentation and configuration (ie add unmock-runner to `lint-ts` and `lint-ts-fix` commands)

@mikesol My build is still failing because it cannot find the `unmock-runner` module. Assuming this is a lerna thing? Any tips for how to resolve would be appreciated.

### CODE REVIEW

**Comment by carolstran** [comment]
Makes sense @mikesol, although I also think it's something that will soon be replaced (as in the next day or two) so maybe not urgent for the time being.

Any thoughts on the build error though? Because that is blocking for merge.

**Comment by mikesol** [comment]
Agreed that it is bikeshedding.
The most recent build I can see is 3260, and in there, the issue is this line:

```typescript
import runner, { IMeeshkanDoneCallback, IRunnerOptions } from "unmock-runner";
```

I think it should be this instead (haven't tested...):

```typescript
import runner, { IMeeshkanDoneCallback, IRunnerOptions } from "./";
```

**Comment by carolstran** [comment]
Everything's passing now! @mikesol, let me know if I should merge.

**Comment by mikesol** [comment]
Cool! @ksaaskil can also add his review, after which my recommendation would be the following:

- merge
- release
- build an `unmock-jest-runner` package that is the `jestRunner`, test and release a `0.0.0` version on npm
- delete the `jestRunner` and release a new `unmock-js`
- update the documentation, README.md, koans, unmock-examples and unmock-katacoda to use `unmock-jest-runner`
- build an `unmock-mocha-runner` package. this is the one that IMO is the most perilous, as we have not yet tested if the abstraction in `unmock-runner` is powerful enough to handle what mocha needs. For example, mocha may have some error validation scheme or return values from tests that require the abstraction in `unmock-runner` (ie the types) to change. In the ideal case, though, it'll be as short as the `jestRunner` and everything will work.

File: packages/unmock-node/src/__tests__/runner.test.ts [hunk 1]
```diff
@@ -1,16 +1,17 @@
 import axios from "axios";
 import * as path from "path";
 import { Service, transform, UnmockPackage } from "unmock-core";
-import { runner } from "..";
 import NodeBackend from "../backend";
 const { withCodes } = transform;
+import jestRunner from "../../../unmock-runner/src/jestRunner";
```
Comment (ksaaskil) [inline packages/unmock-node/src/__tests__/runner.test.ts]: Hmm, I don't think one should refer to other Lerna packages via "../../../" but with something like `import { runner} from "unmock-runner"`. Jest will figure out that `unmock-runner` should point to the local package.

**Comment by carolstran** [comment]
@mikesol Will the `unmock-runner` be accessible via the `unmock-js` package or do we need to create its own npm package? Asking because some of the types/initial setup from the runner is necessary for `unmock-jest-runner`

@ksaaskil That's a good point and I do agree, but it's going to be replaced as soon as we implement the `unmock-jest-runner` (hopefully today or tomorrow). However, if it really bothers you - I can fix it now!

**Comment by ksaaskil** [comment]
> @mikesol Will the `unmock-runner` be accessible via the `unmock-js` package or do we need to create its own npm package? Asking because some of the types/initial setup from the runner is necessary for `unmock-jest-runner`
> 
> @ksaaskil That's a good point and I do agree, but it's going to be replaced as soon as we implement the `unmock-jest-runner` (hopefully today or tomorrow). However, if it really bothers you - I can fix it now!

No reason to fix it now, it was just a comment :)

**Comment by mikesol** [comment]
> @mikesol Will the unmock-runner be accessible via the unmock-js package or do we need to create its own npm package? Asking because some of the types/initial setup from the runner is necessary for unmock-jest-runner
> @ksaaskil That's a good point and I do agree, but it's going to be replaced as soon as we implement the unmock-jest-runner (hopefully today or tomorrow). However, if it really bothers you - I can fix it now!

`unmock-runner` will be its own `npm` package that we'll import from, and it will be published automatically by `lerna`. The types will be accessible from that package. `unmock-js` is not a package AFAIK. Does that answer your question?

**Comment by carolstran** [comment]
@mikesol Yes it does, thank you! Didn't realize that lerna publishes packages automatically.

**Pull request merged**